### ISSUE TITLE
Crash when opening a "too old" message found by fulltext search

### ISSUE THREAD

**Issue by Natanji** [signalapp/Signal-Android#8145]
- [x] I have searched open and closed issues for duplicates
- [x] I am submitting a bug report for existing functionality that does not work as intended
- [x] I have read https://github.com/signalapp/Signal-Android/wiki/Submitting-useful-bug-reports
- [x] This isn't a feature request or a discussion topic

----------------------------------------

### Bug description
Signal crashes every time that I search for an older message (using the relatively new fulltext search function), and then tap on the message. I suspect the root cause for this might be the same as for #4667, but because it doesn't just happen when manually scrolling up but also when viewing a message directly from search, plus because that bug report was closed years ago, I am opening a new bug report for this specific behaviour.

As far as I can tell, the criterion for the message age for which the bug happens is that it's not immediately loaded into the conversation in which it appears when opening said conversation normally (without search), i.e. where the message is so old that in order to see it you would have to scroll further "up" than the default scrollable text in the conversation, which I think displays only the last 500 messages.

This happens in all conversations, regardless of whether it's a group chat or not, as far as I can tell.

### Steps to reproduce
- Do a fulltext search from the Signal main screen, for a search term you know appears on an older message. 
- tap on the search result of that older message

**Actual result:** Signal crashes
**Expected result:** Signal displays the message that was found by fulltext search

### Device info
**Device:** Oneplus 3
**Android version:** 8.0.0
**Signal version:** 4.24.1

### Link to debug log
https://debuglogs.org/046a9e96d9bd268941e7ba9d7cc0816a00e6d485d29dd4d906aaa52e06ef466d

**Comment by greyson-signal** [comment]
So I've reproduced a limited version of this crash and fixed it. Specifically, if you searched for message 500 exactly, it would crash due to an off-by-one error I found. However, I haven't been able to reproduce this in a more general sense. Meaning, I can search for super old messages past the initial load limit, and I'm navigated to them as expected.

Does this happen for many older messages, or just one in particular?
